ceph/ceph_admin.py

```diff
@@ -497,59 +545,132 @@ def ceph_monitoring(self, daemon, nodes, **kwargs):
-                                                            client.ip_address))
-                name = 'client.{}'.format(client.hostname)
-                client_keyring = '/etc/ceph/{}.{}.keyring'.format(self.cluster_name,
-                                                                  name)
+                logger.info(
+                    "setting up client on %s-%s" % (client.hostname, client.ip_address)
+                )
+                name = "client.{}".format(client.hostname)
+                client_keyring = "/etc/ceph/{}.{}.keyring".format(
+                    self.cluster_name, name
+                )
 
                 out, _ = self.shell(
                     remote=self.installer,
                     args=[
-                        'ceph', 'auth', 'get-or-create', name,
+                        "ceph",
+                        "auth",
+                        "get-or-create",
+                        name,
                         'mon "allow *"',
                         'osd "allow *"',
                         'mds "allow *"',
-                        'mgr "allow *"'
-                    ]
+                        'mgr "allow *"',
+                    ],
                 )
 
-                client.exec_command(cmd='sudo mkdir -p /etc/ceph')
-                client.exec_command(cmd='sudo chmod 777 /etc/ceph')
+                client.exec_command(cmd="sudo mkdir -p /etc/ceph")
+                client.exec_command(cmd="sudo chmod 777 /etc/ceph")
 
                 keyring_file = client.write_file(
                     sudo=True,
                     file_name=client_keyring,
-                    file_mode='w',
+                    file_mode="w",
                 )
                 keyring_file.write(out)
                 keyring_file.flush()
 
-                client.exec_command(cmd=f'sudo chmod 0644 {client_keyring}')
+                client.exec_command(cmd=f"sudo chmod 0644 {client_keyring}")
 
         elif op == "purge":
             # todo: Handle mgr purge
             raise NotImplementedError
 
+    @nodes_is_empty
     def ceph_iscsi(self, nodes, **kwargs):
         """
         Manage iscsi
         Args:
             nodes: iscsi node list
-            kwargs: key-value arguments
+            kwargs: key-value arguments (future-use)
+
+        kwargs:
+            op: "create|purge" # (default: create, required: true)
         """
-        # todo: Handle iscsi
-        raise NotImplementedError
+        op = kwargs.get("op", "create").lower()
+
+        nodes = nodes if isinstance(nodes, list) else [nodes]
+        if op == "create":
+            iscsis = []
+            daemons = []
+            for node in nodes:
+                logger.info("adding iscsi on %s-%s" % (node.hostname, node.ip_address))
+                iscsis.append("{host}=iscsi.{host}".format(host=node.hostname))
+                daemons.append("iscsi.{host}".format(host=node.hostname))
+
+            if iscsis:
```

Comment:
Since the list cannot be empty... with the `deco`, I think this check can be avoided.